### ISSUE TITLE
Pointer#order method not working

### ISSUE THREAD

**Issue by chrisseaton** [ffi/ffi#813]
Am I using `Pointer#order` correctly? I can't understand these results:

```ruby
require 'ffi'

pointer = FFI::MemoryPointer.from_string("foobar")
p pointer.order                 # => :little
pointer.order :big
p pointer.order                 # => :little
pointer.order :network
p pointer.order                 # => :little

pointer = FFI::Pointer.new(123456)
p pointer.order                 # => :little
pointer.order :big
p pointer.order                 # => :little
pointer.order :network
p pointer.order                 # => :little

pointer = FFI::Pointer.new(123456)
pointer.order :not_a_byte_order_but_no_error
```

My motivation is I want to be able to read for example an `sint32` from memory that is network-endian, on my little-endian machine.

FFI 1.13.1, MRI 2.7.1.

**Comment by larskanis** [comment]
`pointer.order(:big)` returns a new pointer object with swapping read/write functions. `self` is kept unchanged.

**Comment by chrisseaton** [comment]
Aaaah, thanks!

```ruby
require 'ffi'

pointer = FFI::MemoryPointer.from_string("foobar")
p pointer.order                 # => :little
pointer = pointer.order(:big)
p pointer.order                 # => :big
pointer = pointer.order(:network)
p pointer.order                 # => :big

pointer = FFI::Pointer.new(123456)
p pointer.order                 # => :little
pointer = pointer.order(:big)
p pointer.order                 # => :big
pointer = pointer.order(:network)
p pointer.order                 # => :big

pointer = FFI::Pointer.new(123456)
pointer = pointer.order(:not_a_byte_order_but_no_error)
p pointer.order                 # => :little
```

I should say though that this isn't what the documentation says! So there's one bug.

And the second bug is it silently ignoring mis-spelled symbols, I think, but you might disagree.

**Comment by larskanis** [comment]
This is fixed by #830 .
